tests/test_functional.py

```diff
@@ -361,6 +361,19 @@ def test_save_update_note(self, user, testapp, rodne_cislo):
         assert "Lorem ipsum dolor sit amet,</textarea>" in res.text
         assert Note.query.count() == existing_notes + 1
 
+    @pytest.mark.parametrize("rodne_cislo", ["0010126521", "ahoj", 444])
+    def test_non_exist_rc(self, user, testapp, rodne_cislo):
+        login(user, testapp)
+        res = testapp.get(url_for("donor.detail", rc=rodne_cislo), status=404)
+        test_text = (
+            "<h1>404 Stránka nenalezena</h1>\n"
+            + "  <p>Stránka, kterou hledáte neexistuje</p>\n"
+            + '  <p><a href="/">Domů</a></p>\n'
+        )
+
+        assert res.status_code == 404
+        assert test_text in res.text
```

Comment:
```suggestion
        assert "404 Stránka nenalezena" in res.text
```

Tohle by snad mohlo stačit. Nepotřebujeme úplně testovat celou šablonu.